src/Bicep.Core/Extensions/IFileHandleExtensions.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Security;
+using System.Text;
+using System.Threading.Tasks;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Utils;
+using Bicep.IO.Abstraction;
+
+namespace Bicep.Core.Extensions
+{
+    public static class IFileHandleExtensions
+    {
+        public static bool IsArmTemplateLikeFile(this IFileHandle fileHandle) =>
+                fileHandle.Uri.HasExtension(LanguageConstants.JsonFileExtension) ||
+                fileHandle.Uri.HasExtension(LanguageConstants.JsoncFileExtension) ||
+                fileHandle.Uri.HasExtension(LanguageConstants.ArmTemplateFileExtension);
+
+        public static bool IsBicepFile(this IFileHandle fileHandle) => fileHandle.Uri.HasExtension(LanguageConstants.LanguageFileExtension);
```

Comment:
>IsBicepFile [](http://example.com/codeflow?start=27&length=11)

Do we need to worry about weird cases like directory paths that end with .bicep?

Reply:
We don't need to handle this case here. If a directory's path ends with .bicep and an attempt is made to read or write it, the error will be handled appropriately.

Reply:
Sounds good